include/SFML/Graphics/StencilMode.hpp

```diff
@@ -0,0 +1,210 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2023 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#pragma once
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Graphics/Export.hpp>
+
+
+namespace sf
+{
+
+////////////////////////////////////////////////////////////
+/// \brief Stencil modes for drawing
+///
+////////////////////////////////////////////////////////////
+struct SFML_GRAPHICS_API StencilMode
+{
+    ////////////////////////////////////////////////////////
+    /// \brief Enumeration of the stencil test comparisons that can be performed
+    ///
+    /// The comparisons are mapped directly to their OpenGL equivalents,
+    /// specified by glStencilFunc().
+    ////////////////////////////////////////////////////////
+    enum class Comparison
```

Comment:
Is the word "comparison" the usual word in that context? Wouldn't "condition" or "test" for example be more correct?

Reply:
The "test" refers to the operation that happens at a specific pipeline stage. It is what is done. Condition could imply that arbitrary conditions can be tested. Comparison describes what is actually performed as part of the test: a comparison of 2 integer values.

See: https://registry.khronos.org/OpenGL-Refpages/gl4/html/glStencilFunc.xhtml where the term comparison is also used to describe the technical implementation of the test.

Reply:
Thank you for the explanation. I was asking because Never and Always do not look like comparisons to me. Anyways it is used in OpenGL documentation, so I guess we can keep it.